com.lettier.movie-monad.json

```diff
@@ -0,0 +1,65 @@
+{
+  "app-id": "com.lettier.movie-monad",
+  "runtime": "org.gnome.Platform",
+  "runtime-version": "3.26",
+  "sdk": "org.gnome.Sdk",
+  "command": "movie-monad",
+  "rename-desktop-file": "movie-monad.desktop",
+  "rename-icon": "movie-monad-icon",
+  "rename-appdata-file": "movie-monad.appdata.xml",
+  "finish-args": [
+    "--share=ipc",
+    "--socket=x11",
+    "--socket=wayland",
+    "--device=dri",
+    "--share=network",
+    "--socket=pulseaudio",
+    "--talk-name=org.gtk.vfs",
+    "--talk-name=org.gtk.vfs.*",
+    "--talk-name=org.gnome.SettingsDaemon",
+    "--talk-name=org.gnome.SettingsDaemon.MediaKeys",
+    "--env=DCONF_USER_CONFIG_DIR=.config/dconf",
+    "--filesystem=host",
+    "--talk-name=ca.desrt.dconf",
+    "--env=SSL_CERT_FILE=/etc/ssl/certs/ca-certificates.crt"
+  ],
+  "modules": [
+    {
+      "name": "movie-monad",
+      "buildsystem": "simple",
+      "build-commands": [
+        "mv bin /app",
```

Comment:
You should be building this from source. I realize that is probably easier said than done since no other Haskell Flatpaks exist..

Reply:
Actually incorrect! We have Bustle https://github.com/flathub/org.freedesktop.Bustle/blob/master/org.freedesktop.Bustle.json

Reply:
Note that building from source from within the sand-boxed build environment is not necessary given that the released pre-built binaries are fully compatible with the specified architecture and run time. Given that the run time environment is sand boxed, there is no risk of incompatible run time dependencies provided the host system can run the sand boxed environment. Nonetheless, the pre-built binaries run on a wide range of non-sand-boxed environments such as Ubuntu 14.04 or Deepin 15.4.1. The only real hard requirement is a x86-64 architecture, Linux based distribution capable of running GTK+ 3.10 or greater. In the Flatpak example provided, not every dependency is built from source. The Haskell Platform is already pre-built and merely copied over. 

The only real effect of changing the manifest to build from source is a substantial increase in build time.

:+1:

Reply:
> Note that building from source from within the sand-boxed build environment is not necessary given that the released pre-built binaries are fully compatible with the specified architecture and run time.

Even if we assume ABI is perfectly compatible (which is not a good idea to ever assume). There are benefits to always building it such as verifying the source for security, being able to patch it, etc.